pkg/apis/duck/v1alpha1/channel_template_types.go

```diff
@@ -0,0 +1,45 @@
+/*
+Copyright 2019 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import "k8s.io/apimachinery/pkg/runtime"
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+type ChannelTemplateSpec struct {
+	metav1.TypeMeta `json:",inline"`
+
+	// Spec defines the Spec to use for each channel created. Passed
+	// in verbatim to the Channel CRD as Spec section.
+	// +optional
+	Spec *runtime.RawExtension `json:"spec,omitempty"`
+}
+
+// Internal version of ChannelTemplateSpec that includes ObjectMeta so that
+// we can easily create new Channels off of it.
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+type ChannelTemplateSpecInternal struct {
```

Comment:
Does this show up as a public API? If so, can we move it someplace where it is not exposed to users in the same place as our actual API objects.

Reply:
I guess so. It's being used from apis/eventing and apis/messaging. It was previously duplicated in those places, so I ended up moving it "up" to duck. What other place you think we can put this on?
I think is the same idea as SubscribableType or AddressableType, which are also exposed. 
Do you think that can be a problem?

Reply:
I guess you are right, we have a number of non-real types in the duck API area, this one isn't any worse than the others.

Reply:
OK, will leave it here for now then. Thanks!